File: java/src/main/java/com/penapereira/patterns/bridge/Abstraction.java
package com.penapereira.patterns.bridge;

/** Holds the implementor and forwards the primitive operation to it. */
public class Abstraction {

    protected final Implementor implementor;

    public Abstraction(Implementor implementor) {
        this.implementor = implementor;
    }

    public String operation() {
        return "Abstraction(" + implementor.operationImpl() + ")";
    }
}

File: javascript/src/bridge/abstraction.js
/** Holds the implementor and forwards the primitive operation to it. */
export class Abstraction {
  constructor(implementor) {
    // Not a #private field: RefinedAbstraction needs to reach it.
    this.implementor = implementor;
  }

  operation() {
    return `Abstraction(${this.implementor.operationImpl()})`;
  }
}
